### PR TITLE
Reintroduce timestamp for message.

### PR DESCRIPTION
**Pull request by boaks**
Add waitOnSent for message.

Signed-off-by: Achim Kraus <achim.kraus@bosch-si.com>

### CODE REVIEW

File: californium-core/src/main/java/org/eclipse/californium/core/network/stack/ReliabilityLayer.java [hunk 1]
```diff
@@ -213,7 +213,7 @@ public void run() {
 	public void receiveRequest(final Exchange exchange, final Request request) {
 
 		if (request.isDuplicate()) {
-			if (exchange.getSendNanoTimestamp() > request.getReceiveNanoTimestamp()) {
+			if (exchange.getSendNanoTimestamp() > request.getNanoTimestamp()) {
```
Comment (sbernard31) [inline californium-core/src/main/java/org/eclipse/californium/core/network/stack/ReliabilityLayer.java]: Should we do that  for response too  ? I'm not sure

File: californium-core/src/main/java/org/eclipse/californium/core/network/stack/ReliabilityLayer.java [hunk 2]
```diff
@@ -213,7 +213,7 @@ public void run() {
...
 				// received before response was sent
 				int count = counter.incrementAndGet();
```
Comment (sbernard31) [inline californium-core/src/main/java/org/eclipse/californium/core/network/stack/ReliabilityLayer.java]: I still not get why we count this :sweat_smile:

File: californium-core/src/main/java/org/eclipse/californium/core/network/stack/ReliabilityLayer.java (hunk 1 above)
Comment (boaks) [reply californium-core/src/main/java/org/eclipse/californium/core/network/stack/ReliabilityLayer.java]: Maybe. I try it.

File: californium-core/src/main/java/org/eclipse/californium/core/network/stack/ReliabilityLayer.java (hunk 2 above)
Comment (boaks) [reply californium-core/src/main/java/org/eclipse/californium/core/network/stack/ReliabilityLayer.java]: Just to have a small statistic, if that pays off.

**Comment by sbernard31** [comment]
LGTM.
(Just to let you know if you are using log.debug, this will be less visible for us when we will deploy it)

**Comment by boaks** [comment]
> (Just to let you know if you are using log.debug, this will be less visible for us when we will deploy it)

Therefore I used it for the statistic. My understanding is, that you don't see too much benefit in it.

**Pull request merged**